enterprise/signals.py

```diff
@@ -107,6 +109,22 @@ def _complete_user_enrollment():
         pending_ecu.delete()
 
 
+@receiver(pre_save, sender=EnterpriseCustomer)
+def update_lang_pref_of_all_learners(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update the language preference of all the learners belonging to the enterprise customer.
+    Set the language preference to the value enterprise customer has used as the `default_language`.
+    """
+    # Unset the language preference when a new learner is linked with the enterprise customer.
+    # The middleware in the enterprise will handle the cases for setting a proper language for the learner.
+    if instance.default_language:
+        prev_state = EnterpriseCustomer.objects.filter(uuid=instance.uuid).first()
+        if prev_state is None or prev_state.default_language != instance.default_language:
+            # Unset the language preference when all the learners linked with the enterprise customer.
```

Comment:
nit: Unset the language preference of all the learners linked with the enterprise customer.